### PR TITLE
installed and linked bugsnag, test button on settings page

### PR DESCRIPTION
**Pull request by OSPFNeighbour**
Ive installed and linked Bugsnag, and applied for an opensource free plan with them for the project (they give free use to opensource projects)

https://www.bugsnag.com

@sdunster @cmiceli @sjmcculloch have I linked this in correctly? never done it before TBH.

Test and seems to work, I get bug reports showing in the bugsnag dashboard

### CODE REVIEW

File: client/src/app.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
 import { AsyncStorage, Alert } from 'react-native';
+import { Client } from 'bugsnag-react-native';
```
Comment (sdunster) [inline client/src/app.js]: Suggest you name this something less ambiguous:
```
import { Client as BugsnagClient } from 'bugsnag-react-native';
```

File: client/src/app.js [hunk 2]
```diff
@@ -27,6 +28,8 @@ export const firebaseClient = new FirebaseClient({
   },
 });
 
+export const bugsnag = new Client();
```
Comment (sdunster) [inline client/src/app.js]: protip: put this in it's own file `bugsnag.js` then import from there so it's separate from the rest of the intermingled garbage in `app.js`

**Comment by OSPFNeighbour** [comment]
It's currently set to submit any unhanded react or core errors. That's it's default. Can be set to submit handled and unhandled from either layer.

I'll test what it does with rebuild & also address your points. Thx.

**Comment by sjmcculloch** [comment]
I'm not an expert in this area around the linking area - haven't had to do it much in the past and I've always stayed bound to expo (no ios/android folders). Has it been extensively tested, does the app still function correctly?

**Comment by sdunster** [comment]
This introduced a bunch of project configuration changes unrelated to this diff for firebase cloud messaging that we ultimately decided to remove. I split the parts we want to keep into 2 branches:

* `sdunster.install_bugsnag` #234 
* `sdunster.configure_bugsnag`

**Pull request closed without merging**